designer/client/components/Visualisation/__tests__/Visualisation.jest.tsx

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import { Visualisation } from "../Visualisation";
+import { act, render, waitFor } from "@testing-library/react";
+import { DataContext } from "../../../context";
+import { Data } from "@xgovformbuilder/model";
+
+const customRender = (ui, { providerProps, ...renderOptions }) => {
+  return render(
+    <DataContext.Provider value={providerProps}>{ui}</DataContext.Provider>,
+    renderOptions
+  );
+};
+
+test("Graph is rendered with correct number of pages and updates ", async () => {
```

Comment:
Out of curiosity why are there async/await keywords here? Does this test make API calls in order for it to work?

Reply:
No api calls, but the component needs a tiny bit of time to calculate the nodes of the graph, then it has to draw and render them.

Reply:
Ah cool, I'll keep that in mind if I run into a similar issue when writing tests 👍